### DISCUSSION TITLE
Docs do not explain how to update values in a spreadsheet

### DISCUSSION THREAD

**Discussion by jwmurray** [Byron/google-apis-rs#272] [Q&A] [answered]
The docs at http://byron.github.io/google-apis-rs/google_sheets4_cli/ show the following options for update-values for sheets4-cli:

sheet4 spreadsheets values-update <spreadsheet-id> <range> (-r <kv>)... [-p <v>]... [-o <out>]

I have retrieved values from google sheets (using get-values), but I have not been able to figure out how to update cells.

I have tried the following, 

`./sheets4 spreadsheets values-update 1-Ow1Q01z5z_oSwAa8uG6oP2qUGjjcvbZkFbYt0oGhWg A1:B2,majorDimension=ROWS  -r . [[1,2],[3,4]]`

but only get a  400 error, "bad request"  I am sure that the docs could give a CLI example that provides an update value in a complete command line.  

Thank you!

**Comment by Byron** [comment, marked as answer]
Sorry for the late reply. I believe you could use the `--debug` flag to monitor all communication between client and server (_please be aware that this dumps secrets into the terminal_). That way one could compare between what is [officially documented](https://developers.google.com/sheets/) and the request that gets actually produced.

That would also allow to doctor a curl request which can be manipulated into a request which actually works. From there, maybe it becomes more obvious what the problem is.

As I have never used the sheets API, I can't provide more advise.
Good luck!
